fennel/expr/test_expr.py

```diff
@@ -94,17 +101,31 @@ def test_math_expr():
     assert ret.tolist() == [5, 6, 7, 8]
     assert expr.typeof({"a": float}) == int
 
+    ref_extractor = FetchReferences()
+    ref_extractor.visit(expr.root)
+    assert ref_extractor.refs == {"a"}
+
     expr = (
         when(F("a").num.floor() > 5)
         .then(F("b"))
         .when(F("a") > 3)
         .then(F("a"))
-        .otherwise(1)
+        .otherwise(1 + F("d"))
+    )
+    df = pd.DataFrame(
+        {
+            "a": [1.4, 3.2, 6.1, 4.8],
+            "b": [100, 200, 300, 400],
+            "d": [1, 2, 3, 4],
+        }
     )
-    df = pd.DataFrame({"a": [1.4, 3.2, 6.1, 4.8], "b": [100, 200, 300, 400]})
-    ret = expr.eval(df, {"a": float, "b": int})
-    assert ret.tolist() == [1, 3.2, 300, 4.8]
-    assert expr.typeof({"a": float, "b": int}) == float
+    ret = expr.eval(df, {"a": float, "b": int, "d": int})
+    assert ret.tolist() == [2.0, 3.2, 300, 4.8]
+    assert expr.typeof({"a": float, "b": int, "d": int}) == float
```

Comment:
Unrelated to this PR but since `typeof` only takes schema, why not pass this as kwargs instead of a dictionary? Like

`expr.typeof(a=float, b=int, d=int)`

Don't feel very strongly though, upto you. Maybe we can support both?

Reply:
So that we can pass dataset.schema() to it directly